### ISSUE TITLE
Support MyST markdown

### ISSUE THREAD

**Issue by choldgraf** [sphinx-gallery/sphinx-gallery#710]
There's a relatively new parser for markdown in Sphinx called [the myst-markdown parser](https://myst-parser.readthedocs.io/en/latest/). It allows users to write their documentation in markdown and retain all of the Sphinx features like roles, directives, etc.

Enabling myst markdown in a repository automatically adds the MyST parser for any `.md` files (and it retains the sphinx build's support for `.rst` files).

Currently, we have a hard assumption that the text people write in `sphinx-gallery` `.py` files is in reStructuredText, but it would be great if this also supported markdown in the generate page outputs. This would have a few benefits:

1. We wouldn't need to run an rST to markdown converter when generating ipynb files if the `.py` files already had markdown in them
2. Users could easily convert their pre-existing notebooks into sphinx-gallery `.py` files by using `jupytext`
3. Some users may be more familiar with, or enjoy writing, markdown more-so than reStructuredText

It may be a decent amount of work, I'm not sure (and we'd probably wanna start somewhere in https://github.com/sphinx-gallery/sphinx-gallery/blob/master/sphinx_gallery/gen_rst.py), but I wanted to open this to track the issue

**Comment by lucyleeow** [comment]
I agree that markdown is more well known and notebook conversion could be handy.

I think there would be extensive additions to support this though. To get an idea of what needs to be changed to support markdown, what would be the markdown version of the simple rST example below? (@choldgraf)

```rst
.. only:: html

    .. note::
        :class: sphx-glr-download-link-note

        Click :ref:`here <sphx_glr_download_auto_examples_local_module.py>`     to download the full  code
    .. rst-class:: sphx-glr-example-title

    .. _sphx_glr_auto_examples_local_module.py:

Header
======

Trivial module to provide a value for plot_numpy_matplotlib.py


.. code-block:: python3


    N = 1000


.. rst-class:: sphx-glr-timing

   **Total running time of the script:** ( 0 minutes  0.000 seconds)


.. _sphx_glr_download_auto_examples_local_module.py:


.. only :: html

 .. container:: sphx-glr-footer
    :class: sphx-glr-footer-example



  .. container:: sphx-glr-download sphx-glr-download-python

     :download:`Download Python source code: local_module.py <local_module.py>`



  .. container:: sphx-glr-download sphx-glr-download-jupyter

     :download:`Download Jupyter notebook: local_module.ipynb <local_module.ipynb>`


.. only:: html

 .. rst-class:: sphx-glr-signature

    `Gallery generated by Sphinx-Gallery <https://sphinx-gallery.github.io>`_
```

**Comment by choldgraf** [comment]
Hey - sure thing, here's what it would look like in MyST markdown:

```````
````{only} html

```{note}
:class: sphx-glr-download-link-note
Click {ref}`here <sphx_glr_download_auto_examples_local_module.py>` to
download the full  code
```

```{rst-class} sphx-glr-example-title   <--- I'm not sure what this directive does but it seems rST-specific?
```
````

(sphx_glr_auto_examples_local_module.py)=
# Header

Trivial module to provide a value for plot_numpy_matplotlib.py

```{code-block} python3
N = 1000
```

```{rst-class} sphx-glr-timing
**Total running time of the script:** ( 0 minutes  0.000 seconds)
```

(sphx_glr_download_auto_examples_local_module.py)=
`````{only} html
````{container} spx-glr-footer
:class: sphx-glr-footer-example

```{container} sphx-glr-download sphx-glr-download-python
{download}`Download Python source code: local_module.py <local_module.py>`
```

```{container} sphx-glr-download sphx-glr-download-jupyter
{download}`Download Jupyter notebook: local_module.ipynb <local_module.ipynb>`
```
````

````{only} html
```{rst-class} sphx-glr-signature
[Gallery generated by Sphinx-Gallery](https://sphinx-gallery.github.io)
```
````

`````
```````

(I may be missing a back-tick in there somewhere but this is more-or-less what it would look like in MyST)

**Comment by josh146** [comment]
I'm curious in the current status of this issue - we currently have a large tutorial repository that uses Sphinx Gallery, and it would be great if we could use both ReST and MyST markup across our various tutorials

**Comment by lucyleeow** [comment]
I think it would be a useful addition and MyST is increasingly popular but unfortunately I don't think anyone has the spare cycles to do the work.

**Comment by dhapolapankaj** [comment]
Hello Guys,

I see that we have following code in .rst file, which means it will include the markdown file inside the .rst file. 

`.. include:: action_func.md

     :parser: myst_parser.sphinx_
`

I don't want the separate .md file to maintain and directly put the markdown inside the rst file with some instruction to sphinx that process this part as `[myst-parser](https://github.com/executablebooks/MyST-Parser)` markdown.

Do we have such option?
Any leads will be very helpful.

Thanks
PD

**Comment by Fanchengyan** [comment]
Hi all, I have developed a `MyST-Sphinx-Gallery` that supports the `MyST` ecosystem, which means you can create a gallery using any of the markdown, Jupyter notebook, or rst file formats.
Besides, it supports creating galleries directly calling directives: [Using Directives](https://myst-sphinx-gallery.readthedocs.io/en/latest/user_guide/gen_gallery/directive/index.html)
and [Demo Gallery](https://myst-sphinx-gallery.readthedocs.io/en/latest/examples/index.html)
